pkg/metrics-server/server.go

```diff
@@ -129,40 +128,57 @@ func (ms *MetricsServer) scrape(ctx context.Context, startTime time.Time) {
 	recvErr := ms.storage.Store(data)
 	if recvErr != nil {
 		klog.Errorf("unable to save metrics: %v", recvErr)
-
-		// any failure to save means we're unhealthy
-		healthyTick = false
+		// any failure to save means unhealthy scrape
+		healthyScrape = false
 	}
 
 	collectTime := time.Since(startTime)
 	tickDuration.Observe(float64(collectTime) / float64(time.Second))
 	klog.V(6).Infof("...Cycle complete")
 
 	ms.healthMu.Lock()
-	ms.lastOk = healthyTick
+	ms.lastOk = true
+	ms.healthyScrape = healthyScrape
 	ms.healthMu.Unlock()
+
 }
 
-// CheckHealth checks the health of the manager by looking at tick times,
-// and checking if we have at least one node in the collected data.
-// It implements the health checker func part of the healthz checker.
-func (ms *MetricsServer) CheckHealth(_ *http.Request) error {
+// Check if MS is alive by looking at last tick time and checking if
+// last scrape completed
+// Serves the Liveness probe
+func (ms *MetricsServer) CheckLiveness(_ *http.Request) error {
 	ms.healthMu.RLock()
 	lastTick := ms.lastTickStart
-	healthyTick := ms.lastOk
+	lastOk := ms.lastOk
```

Comment:
Please read definition of probe, scrape doesn't need to succeed for liveness check to succeed. We want to check if scraping happens without distuption.